project-base/storefront/components/Blocks/Product/ProductsSlider.tsx

```diff
@@ -84,6 +95,10 @@ export const ProductsSlider: FC<ProductsSliderProps> = ({
                     <SliderButton type="next" onClick={onMoveToNextSlideHandler} />
                 </div>
             ) : null}
+
+            {isPopupCompareOpen && (
+                <ProductComparePopup isVisible onCloseCallback={() => setIsPopupCompareOpen(false)} />
+            )}
```

Comment:
The popup itself has a `isVisible` prop. Is it not enough? Do we need to wrap it in a ternary operator? I understand that there is some underlying logic inside the component which is not run this way, but I am wondering if that's enough of an added value. Furthermore, if you check the popup component, we are setting a event listener if `isVisible=true` and removing it if `isVisible=false`. This way we never remove it, which I don't think is a good practice.

Reply:
Ah, this is my refactoring code. Good point. I am used to from external libraries which use Popup that they mount Popup anyway and just control by CSS it's visibility. This way Popup is always in the DOM making it larger. So I subconsciously put own control logic without noticing that it's our own component which is mounted/unmounted properly.

Reply:
TBH, I have no idea... 
After consultation with the original author, I rewrote it in 95e5b91465